meerket/meerket-domain/src/main/java/org/j1p5/domain/product/service/ProductService.java

```diff
@@ -100,4 +98,22 @@ public ProductResponseDetailInfo getProductDetail(Long productId, Long userId){
 
     }
 
+    @Transactional
+    public void updateProduct(Long productId, Long userId, ProductUpdateInfo info){
+        UserEntity user = userReader.getUser(userId);
+        ProductEntity product = productRepository.findById(productId)
+                .orElseThrow(() ->new DomainException(PRODUCT_NOT_FOUND));
+
+        if(!(product.getUser().equals(user))){
+            throw new DomainException(PRODUCT_NOT_AUTHORIZED);
+        }
+
+        if(!product.isHasBuyer()){
```

Comment:
```suggestion
        if(product.isHasBuyer()){
            throw new DomainException(PRODUCT_HAS_BUYER);
        }
        product.updateProduct(info);
```
방식이 가독성 측면에서 더 깔끔하고 else없이도 작성할 수 있을 것 같아요.